### PR TITLE
Created git-jira-pr.  Added a Jira module to hold some common functionality.

### PR DESCRIPTION
**Pull request by martyphee**
Standardized the branch naming to [PROJECT_KEY]-[ISSUE_ID]/[SUMMARY]

### CODE REVIEW

File: bin/git-jira-pr [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+#
+require 'rubygems'
+require 'jira'
+require 'optparse'
+require 'cgi'
+require 'term/ansicolor'
+require 'git-whistles/app'
+require 'git-whistles/jira'
+
+class App < Git::Whistles::App
```
Comment (pedrocunha) [inline bin/git-jira-pr]: For the consistency do you mind changing this file to only have 1 empty line inbetween methods?

File: lib/git-whistles/jira.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+module Git
+  module Whistles
+    module Jira
+      def get_config
+        username = `git config jira.username`.strip
+        if username.empty?
+          puts Term::ANSIColor.yellow %Q{
```
Comment (pedrocunha) [inline lib/git-whistles/jira.rb]: The indentation looks a bit from before this being moved. Do you mind keeping it the same please? 

```
puts Term::ANSIColor.yellow %Q{
  Your branch appears to have a issue ID,
  but I don't know your JIRA username!
  Please set it with:
    $ git config [--global] jira.username <username>
}
```

File: lib/git-whistles/jira.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
+module Git
+  module Whistles
+    module Jira
```
Comment (pedrocunha) [inline lib/git-whistles/jira.rb]: I appreciate you took the time to extract common functionality! It is definitely a good idea! 

I personally don't like much module inclusion as it makes test a bit more cucumbersome by having to create extra classes or testing this from somewhere else, and also its a pain when the public methods can clash with others. To complete my comment with some constructive idea: I would rather see this guy as a class with all its methods being class methods.

What do you think about this?

**Comment by pedrocunha** [comment]
Thanks @martyphee for this! Looks good. Hope my comments make sense. Also I think you might have to handle conflicts.

**Comment by martyphee** [comment]
I'm making the updates now.  It was a bit of a rush.  Thank you for the feedback.

**Comment by martyphee** [comment]
@pedrocunha I made the changes and have been using it the last few days.

**Comment by pedrocunha** [comment]
Sorry @martyphee for no feedback on this yet but do you mind solving the conflicts first please. Thanks. I'll try to review & merge this today.

**Comment by martyphee** [comment]
Done, thx.

File: bin/git-jira-pr [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    {
+      :from   => run!('git symbolic-ref HEAD').strip.gsub(%r(^refs/heads/), ""),
+      :to     => 'master',
+      :remote => 'origin'
+    }
+  end
+
+  def option_parser
+    @option_parser ||= OptionParser.new do |op|
+      op.banner = "Usage: git pivotal-pr [options]"
```
Comment (pedrocunha) [inline bin/git-jira-pr]: I think here is `jira-pr` ?

File: bin/git-jira-pr [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+      team, issue_id = $1.capitalize.split('-')
+    else
+      issue_id = team = nil
+    end
+
+    # guess title.
+    title = options.from.split('/').last.split(/[_-]/).delete_if { |word| word =~ /^\d+$/ }.join(' ').capitalize
+    query[:"pull_request[title]"] = team ? "#{team}: #{title}" : title
+
+    # add Pivotal infos
```
Comment (pedrocunha) [inline bin/git-jira-pr]: maybe remove this comment ?

Comment (martyphee) [inline bin/git-jira-pr]: Done, thx for catching that.

**Comment by pedrocunha** [comment]
Cheers @martyphee

**Pull request merged**